kratos/python_scripts/analysis_stage.py

```diff
@@ -202,6 +219,74 @@ def ChangeMaterialProperties(self):
         """this function is where the user could change material parameters as a part of the solution step """
         pass
 
+    def ClearDatabase(self):
+        """ This method clears the database in case it is necessary
+
+            Keyword arguments:
+            self It signifies an instance of a class.
+        """
+        pass
+
+    def ReInitializeSolver(self):
+        """ This reinitializes the solver and the processes (used for example on remesh or on adaptive NR)
+
+            Keyword arguments:
+            self It signifies an instance of a class.
+        """
+        solver = self._GetSolver()
+        processes = self._GetListOfProcesses()
+
+        reinitialize_requirement = self.GetReInitializeRequired()
+
+        # Clear solver
+        solver.Clear()
+
+        # Some initilizations
```

Comment:
All these different options should be implemented in derived analyses stages, not the base one.